### PR TITLE
Extra headers were not encoded

### PR DESCRIPTION
**Pull request by Neku**

### CODE REVIEW

File: src/JWT/IJwtEncoder.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public interface IJwtEncoder
         /// <param name="payload">An arbitrary payload (must be serializable to JSON).</param>
         /// <param name="key">The key bytes used to sign the token.</param>
         /// <returns>The generated JWT.</returns>
-        string Encode(IDictionary<string, object> extraHeaders, object payload, string key);
+        string Encode(IDictionary<string, string> extraHeaders, object payload, string key);
```
[the remark this answers is not in the document]
Comment (Neku) [reply src/JWT/IJwtEncoder.cs]: Ok, I'll try to make it work with the object. I had few issues with that.
I'll update my PR

[the remark this answers is not in the document]
Comment (abatishchev) [reply src/JWT/IJwtEncoder.cs]: Hi, any chance to come back to this? Would be cool. Thanks.

[the remark this answers is not in the document]
Comment (Neku) [reply src/JWT/IJwtEncoder.cs]: I'll be to work on this soon, as I'll resume the linked project

[the remark this answers is not in the document]
Comment (abatishchev) [reply src/JWT/IJwtEncoder.cs]: Please come back to this. 5.0.0-betaX is out, very good timing to add some more features, break something if needed :)

[the remark this answers is not in the document]
Comment (abatishchev) [reply src/JWT/IJwtEncoder.cs]: Please come back to this.

**Comment by pkolanu66** [comment]
Hi,

Can anyone please tell me the difference between writing customized JWTs(I assume that we are doing with this project) code and the System.IdentityModel.Tokens namespace ?

File: src/JWT/IJwtEncoder.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (abatishchev) [reply src/JWT/IJwtEncoder.cs]: Ping

**Comment by abatishchev** [comment]
Better later than never :)

**Pull request merged**